src/test/ram_statement_equal_clone_test.cpp

```diff
@@ -0,0 +1,470 @@
+    /*
+     * QUERY
+     *  FOR t0 IN A
+     *   PROJECT (t0.0, t0.2) INTO B
+     */
+    std::vector<std::unique_ptr<RamExpression>> a_expressions;
+    a_expressions.emplace_back(new RamTupleElement(0, 0));
+    a_expressions.emplace_back(new RamTupleElement(0, 2));
+    auto a_project = std::make_unique<RamProject>(
+            std::make_unique<RamRelationReference>(&B), std::move(a_expressions));
+    auto a_scan = std::make_unique<RamScan>(
+            std::make_unique<RamRelationReference>(&A), 0, std::move(a_project), "");
+
+    std::vector<std::unique_ptr<RamExpression>> b_expressions;
+    b_expressions.emplace_back(new RamTupleElement(0, 0));
+    b_expressions.emplace_back(new RamTupleElement(0, 2));
+    auto b_project = std::make_unique<RamProject>(
+            std::make_unique<RamRelationReference>(&B), std::move(b_expressions));
+    auto b_scan = std::make_unique<RamScan>(
+            std::make_unique<RamRelationReference>(&A), 0, std::move(b_project), "");
+
+    RamQuery a(std::move(a_scan));
+    RamQuery b(std::move(b_scan));
+    EXPECT_EQ(a, b);
+    EXPECT_NE(&a, &b);
+
+    RamQuery* c = a.clone();
+    EXPECT_EQ(a, *c);
+    EXPECT_NE(&a, c);
+    delete c;
+
+    /*
+     * QUERY
+     *  PARALLEL CHOICE t1 IN A WHERE (t1.0 = 0)
+     *   RETURN (t1.2)
+     */
+    std::vector<std::unique_ptr<RamExpression>> d_return_value;
+    d_return_value.emplace_back(new RamTupleElement(1, 0));
+    auto d_return = std::make_unique<RamSubroutineReturnValue>(std::move(d_return_value));
+    // condition t1.0 = 0
+    auto d_cond = std::make_unique<RamConstraint>(
+            BinaryConstraintOp::EQ, std::make_unique<RamTupleElement>(1, 0), std::make_unique<RamNumber>(0));
+    auto d_parallel_choice = std::make_unique<RamParallelChoice>(
+            std::make_unique<RamRelationReference>(&A), 1, std::move(d_cond), std::move(d_return), "");
+
+    std::vector<std::unique_ptr<RamExpression>> e_return_value;
+    e_return_value.emplace_back(new RamTupleElement(1, 0));
+    auto e_return = std::make_unique<RamSubroutineReturnValue>(std::move(e_return_value));
+    // condition t1.0 = 0
+    auto e_cond = std::make_unique<RamConstraint>(
+            BinaryConstraintOp::EQ, std::make_unique<RamTupleElement>(1, 0), std::make_unique<RamNumber>(0));
+    auto e_parallel_choice = std::make_unique<RamParallelChoice>(
+            std::make_unique<RamRelationReference>(&A), 1, std::move(e_cond), std::move(e_return), "");
+    RamQuery d(std::move(d_parallel_choice));
+    RamQuery e(std::move(e_parallel_choice));
+    EXPECT_EQ(d, e);
+    EXPECT_NE(&d, &e);
+
+    RamQuery* f = d.clone();
+    EXPECT_EQ(d, *f);
+    EXPECT_NE(&d, f);
+    delete f;
+}
+
+TEST(RamSequence, CloneAndEquals) {
+    // no statements in the sequence
+    RamSequence a;
+    RamSequence b;
+    EXPECT_EQ(a, b);
+    EXPECT_NE(&a, &b);
+
+    RamSequence* c = a.clone();
```

Comment:
A better name (for this and similar in other tests) might be `aClone` since its purpose is to be a clone of `a`